### PR TITLE
chore: clean up subzones code

### PR DESCRIPTION
**Pull request by aljo242**
Clean up subzones code by renaming helpers:
- `zone.ChainID()` -> `zone.BaseChainID()`
- `zone.ID()` -> `zone.ZoneID()` (redundant i know, but it is clearer.)

Refactor functions that receive ID strings to receive `*Zone` and get the ID string internally.  This way, there are significantly less ID function calls throughout the codebase which was getting confusing.

### CODE REVIEW

File: x/interchainstaking/types/accounts.go [hunk 1]
```diff
@@ -42,3 +42,19 @@ func (a *ICAAccount) DecrementBalanceWaitgroup() error {
 	a.BalanceWaitgroup--
 	return nil
 }
+
+func (z *Zone) DepositPortOwner() string {
+	return z.ZoneID() + ".deposit"
```
Comment (joe-bowman) [inline x/interchainstaking/types/accounts.go:47]: These suffixes exist as constants already.

**Review by joe-bowman — CHANGES_REQUESTED** [review]
Just the suffix consts, else lgtm

File: x/interchainstaking/types/accounts.go (hunk 1 above)
Comment (aljo242) [reply x/interchainstaking/types/accounts.go]: https://github.com/ingenuity-build/quicksilver/pull/502/commits/f22910740f719ccde6c6a1c6d354a96bf6cab222

**Review by joe-bowman — DISMISSED** [review]
lgtm, ty

**Review by joe-bowman — APPROVED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
lgtm

**Pull request merged**